src/HttpAdapterFactory.php

```diff
@@ -81,4 +95,32 @@ public static function register($name, $class)
 
         self::$classes[$name] = $class;
     }
+
+    /**
+     * guesses the best matching adapter
+     *
+     * @param  array                $preferred
+     * @return HttpAdapterInterface
+     * @throws HttpAdapterException
+     */
+    public static function guess($preferred = array())
+    {
+        $available = function ($class) {
+            return class_exists($class) || function_exists($class) || ini_get($class);
+        };
+
+        foreach ($preferred as $preference) {
```

Comment:
Can you cast the `$preferred` as array. That will allow to pass a strint to the `guess` method.